src/shaders/symbol_sdf.fragment.glsl

```diff
@@ -50,9 +55,53 @@ void main() {
         alpha = min(smoothstep(halo_edge - gamma_scaled, halo_edge + gamma_scaled, dist), 1.0 - alpha);
     }
 
-    fragColor = color * (alpha * opacity * fade_opacity);
+    return color * (alpha * opacity * fade_opacity);
+}
+
+vec4 renderMSDFIcon(vec4 fill_color, vec4 halo_color, float opacity, float EDGE_GAMMA, vec2 tex, float gamma_scale, float size, float fade_opacity) {
+
+    float fontScale = size;
+
+    lowp vec4 color = fill_color;
+    highp float gamma = EDGE_GAMMA / (fontScale * u_gamma_scale);
+    lowp float inner_edge = (256.0 - 64.0) / 256.0;
+
+    vec3 s = texture(u_texture, tex).rgb;
+    float sd = median(s);
+    float dist = sd - 0.5;
+
+    if (u_is_halo) {
+        color = halo_color;
+        s = texture(u_texture_2, tex).rgb;
+        sd = median(s);
+        dist = sd - 0.5;
+    }
+
+    float clampedDistance = clamp(dist * screenPxRange(fontScale) + 0.5, 0.0, 1.0);
+    
+    highp float gamma_scaled = gamma * gamma_scale;
+    highp float alpha = smoothstep(inner_edge - gamma_scaled, inner_edge + gamma_scaled, clampedDistance);
+
+    return color * (alpha * opacity * fade_opacity);
+}
+
+void main() {
+    #pragma mapbox: initialize highp vec4 fill_color
+    #pragma mapbox: initialize highp vec4 halo_color
+    #pragma mapbox: initialize lowp float opacity
+    #pragma mapbox: initialize lowp float halo_width
+    #pragma mapbox: initialize lowp float halo_blur
+
+    float EDGE_GAMMA = 0.105 / u_device_pixel_ratio;
+
+    vec2 tex = v_data0.xy;
+    float gamma_scale = v_data1.x;
+    float size = v_data1.y;
+    float fade_opacity = v_data1[2];
+
+    fragColor = u_is_text || !u_is_msdf ? renderText(u_is_text, fill_color, halo_color, opacity, halo_width, halo_blur, EDGE_GAMMA, tex, gamma_scale, size, fade_opacity) : renderMSDFIcon(fill_color, halo_color, opacity, EDGE_GAMMA, tex, gamma_scale, size, fade_opacity);
```

Comment:
I expressed concern about this branching logic in our 1:1, but I think since the input to the branch is a uniform the execution only ever has to follow one path and there ends up not being any efficiency hit.